src/logics/ArithLogic.cc

```diff
@@ -372,11 +372,12 @@ PTRef ArithLogic::mkPlus(vec<PTRef> && args)
     args.clear();
     SymRef s_new;
     simp.simplify(plusSym, flattened_args, s_new, args);
-    if (args.size() == 1)
-        return args[0];
     if (s_new != getPlusForSort(returnSort)) {
         return mkFun(s_new, std::move(args));
     }
+    if (args.size() == 1) {
```

Comment:
Now I am thinking this might cause problems if sum simplifies to a multiplication?!

Reply:
No, it seems the code handles this case correctly.